src/domain/navigation/index.js

```diff
@@ -105,11 +105,35 @@ export function addPanelIfNeeded(navigation, path) {
 }
 
 export function parseUrlPath(urlPath, currentNavPath, defaultSessionId) {
+    const segments = [];
+
+    // Special case for OIDC callback
+    if (urlPath.includes("state")) {
+        const params = new URLSearchParams(urlPath);
+        if (params.has("state")) {
+            // This is a proper OIDC callback
+            if (params.has("code")) {
+                segments.push(new Segment("oidc-callback", [
```

Comment:
why not use an object as a value here rather than an array, then you can pass it straight to _showLogin in RootViewModel...

Reply:
Done!